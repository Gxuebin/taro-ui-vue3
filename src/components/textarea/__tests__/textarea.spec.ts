import { mount } from '@vue/test-utils'
import AtTextarea from '../index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtTextarea as any, {
    slots,
    props: { value: '', onChange: () => { }, ...values },
  })
}

describe('AtTextarea', () => {
  it('should render default AtTextarea', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- value', () => {
    const wrapper = factory({ value: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- maxLength', () => {
    const wrapper = factory({ maxLength: 300 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- placeholder', () => {
    const wrapper = factory({ placeholder: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- count', () => {
    const wrapper = factory({ count: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render prop -- height', () => {
    const wrapper = factory({ height: 3000 })
    expect(wrapper.vm.height).toBe(3000)
    expect(wrapper.element).toMatchSnapshot()
  })
})
