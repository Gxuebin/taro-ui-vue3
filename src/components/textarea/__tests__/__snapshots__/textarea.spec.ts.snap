// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTextarea should render default AtTextarea 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    autofocus="false"
    class="at-textarea__textarea"
    cursorspacing="100"
    fixed="false"
    maxlength="200"
    placeholder=""
    placeholderclass="[object Object]"
    placeholderstyle=""
    showconfirmbar="false"
    style="height: 2.1334rem;"
  />
  <view
    class="at-textarea__counter"
  >
    0 / 200
  </view>
</view>
`;

exports[`AtTextarea should render prop -- class 1`] = `
<view
  class="test at-textarea at-textarea--WEB test"
>
  <textarea
    autofocus="false"
    class="at-textarea__textarea"
    cursorspacing="100"
    fixed="false"
    maxlength="200"
    placeholder=""
    placeholderclass="[object Object]"
    placeholderstyle=""
    showconfirmbar="false"
    style="height: 2.1334rem;"
  />
  <view
    class="at-textarea__counter"
  >
    0 / 200
  </view>
</view>
`;

exports[`AtTextarea should render prop -- count 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    autofocus="false"
    class="at-textarea__textarea"
    cursorspacing="100"
    fixed="false"
    maxlength="200"
    placeholder=""
    placeholderclass="[object Object]"
    placeholderstyle=""
    showconfirmbar="false"
    style="height: 2.1334rem;"
  />
  <!---->
</view>
`;

exports[`AtTextarea should render prop -- height 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    autofocus="false"
    class="at-textarea__textarea"
    cursorspacing="100"
    fixed="false"
    maxlength="200"
    placeholder=""
    placeholderclass="[object Object]"
    placeholderstyle=""
    showconfirmbar="false"
    style="height: 64rem;"
  />
  <view
    class="at-textarea__counter"
  >
    0 / 200
  </view>
</view>
`;

exports[`AtTextarea should render prop -- maxLength 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    autofocus="false"
    class="at-textarea__textarea"
    cursorspacing="100"
    fixed="false"
    maxlength="300"
    placeholder=""
    placeholderclass="[object Object]"
    placeholderstyle=""
    showconfirmbar="false"
    style="height: 2.1334rem;"
  />
  <view
    class="at-textarea__counter"
  >
    0 / 300
  </view>
</view>
`;

exports[`AtTextarea should render prop -- placeholder 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    autofocus="false"
    class="at-textarea__textarea"
    cursorspacing="100"
    fixed="false"
    maxlength="200"
    placeholder="test"
    placeholderclass="[object Object]"
    placeholderstyle=""
    showconfirmbar="false"
    style="height: 2.1334rem;"
  />
  <view
    class="at-textarea__counter"
  >
    0 / 200
  </view>
</view>
`;

exports[`AtTextarea should render prop -- style 1`] = `
<view
  class="at-textarea at-textarea--WEB"
  style="color: red;"
>
  <textarea
    autofocus="false"
    class="at-textarea__textarea"
    cursorspacing="100"
    fixed="false"
    maxlength="200"
    placeholder=""
    placeholderclass="[object Object]"
    placeholderstyle=""
    showconfirmbar="false"
    style="height: 2.1334rem;"
  />
  <view
    class="at-textarea__counter"
  >
    0 / 200
  </view>
</view>
`;

exports[`AtTextarea should render prop -- value 1`] = `
<view
  class="at-textarea at-textarea--WEB"
>
  <textarea
    autofocus="false"
    class="at-textarea__textarea"
    cursorspacing="100"
    fixed="false"
    maxlength="200"
    placeholder=""
    placeholderclass="[object Object]"
    placeholderstyle=""
    showconfirmbar="false"
    style="height: 2.1334rem;"
  />
  <view
    class="at-textarea__counter"
  >
    4 / 200
  </view>
</view>
`;
